drgn_tools/cpuinfo.py

```diff
@@ -0,0 +1,615 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to view cpuinfo data
+"""
+import argparse
+from typing import Any
+from typing import Dict
+
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux.percpu import per_cpu
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.table import print_dictionary
+
+
+def x86_get_cpu_info(prog: Program) -> Dict[str, Any]:
+    """
+    Helper to get cpuinfo data for x86
+
+    :returns: a dictionary of the cpuinfo data
+    """
+    # TODO: Add aarch64 support.
+    # `struct cpuinfo_arm64` being fetched using 'boot_cpu_data' only has
+    # register values, while the struct fetched using 'cpu_data' has all values
+    #  zero/null
+
+    cpus = int(prog["nr_cpu_ids"])
+
+    if "cpu_data" in prog:
+        cpuinfo_struct = prog["cpu_data"]
+    elif "boot_cpu_data" in prog:
+        cpuinfo_struct = prog["boot_cpu_data"]
```

Comment:
Let's use `.read_()` here ([docs](https://drgn.readthedocs.io/en/latest/api_reference.html#drgn.Object.read_)).

The reasoning here is that the cpuinfo struct is a large-ish structure, and we're going to be reading parts of it very frequently. The way drgn handles this (in the live case) is by calling `read(/proc/kcore)` for each time we access a value. And reading from /proc/kcore has a quite high overhead for each call (but the overhead for each additional byte of data read is much lower). Thus, it makes sense to read chunks of data at a time, rather than small reads. If we call `.read_()` like shown, then the whole structure gets read at once, and the result is saved as the value of the object.

This means that the value won't be updated to reflect changes over time. But that also is good, because the chances are much better that your read would be atomic, in the live case.
```suggestion
    if "cpu_data" in prog:
        cpuinfo_struct = prog["cpu_data"].read_()
    elif "boot_cpu_data" in prog:
        cpuinfo_struct = prog["boot_cpu_data"].read_()
```